src/chainparams.cpp

```diff
@@ -897,6 +897,7 @@ class CCustomParams : public CRegTestParams {
         consensus.start_p2wsh_script = args.GetIntArg("-con_start_p2wsh_script", consensus.start_p2wsh_script);
         accept_discount_ct = args.GetBoolArg("-acceptdiscountct", false);
         create_discount_ct = args.GetBoolArg("-creatediscountct", false);
+        if (create_discount_ct) accept_discount_ct = true;
```

Comment:
In 7fc96512ff71c5fea9fb167403c64ac42af1070b:

This is really just a nit, but I think it'd be clearer if you changed `create_discount_ct` to be `accept_discount_ct || args.GetBoolArg("-creatediscountct", false)` or even just `args.GetBoolArg("-creatediscountct", accept_discount_ct);`.

The current code modifies variables twice when conceptually they should be immutable, which makes it easy for readers to miss the second assignment. Especially when it's hidden at the end of an `if` line.

Reply:
done in 523c98d69ff416ec8e61a878b477b9f2c37171f8